testgrid/cmd/summarizer/response/types.proto

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+syntax = "proto3";
+option go_package = "response";
+
+import "testgrid/config/config.proto";
+import "testgrid/summary/summary.proto";
+
+// PhaseTimer times the duration of each update cycle.
+message PhaseTimer {
+  int32 start = 1;
+  int32 last = 2;
+  repeated string phases = 3;
+  repeated double delta = 4;
+  double total = 5;
+}
+
+message TestAlert {
+  string fail_build_id = 1 [json_name="fail-build-id"];
+  int64 fail_count = 2 [json_name="fail-count"];
+  int32 fail_time = 3 [json_name="fail-time"];
+  string text = 4;
+  string message = 5;
+  string link_text = 6 [json_name="link-text"];
+  string link = 7;
+  string url_text = 8 [json_name="url-text"];
+  string test_id = 9 [json_name="test-id"];
+  string pass_build_id = 10 [json_name="pass-build-id"];
+  int64 pass_count = 11 [json_name="pass-count"];
+  int32 pass_time = 12 [json_name="pass-time"];
+  string code_search_path = 13 [json_name="code-search-path"];
+  string test_name = 14 [json_name="test-name"];
+}
+
+// RleStatus represents a run-length encoded test status:
+// a run of Count cells with status Value.
+message RleStatus {
+  int32 count = 1;
+  int32 value = 2;
+}
+
+message RawMetric {
+  // Name.
+  string id = 1;
+  // A sparse encoding of densely stored values. The layout encodes the cycle
+  // indices of the values in the value field below. The layout contains indices
+  // followed by counts. The indices specify test cycles where contiguous
+  // sequences of values start.
+  repeated int32 layout = 2;
+  // Non-empty values for each test result.
+  repeated double value = 3;
+}
+
+message Graph {
+  string metric = 1;
+  repeated string names = 2;
+  repeated string labels = 3;
+
+  message Values {
+    repeated double values = 1;
+  }
+
+  repeated Values values = 4;
+}
+
+// Row describes a test row.
+message Row {
+  string name = 1;
+  string original_name = 2 [json_name="original-name"];
+  TestAlert alert = 3;
+  repeated string linked_bugs = 4;
+  repeated string messages = 5;
+  repeated string short_texts = 6;
+  repeated string test_ids = 7;
+  repeated int64 short_test_ids = 8;
+  repeated RleStatus statuses = 9;
+  string target = 10;
+  repeated Row tests = 11;
+  repeated string tabular_name_groups = 12 [json_name="tabular-name-groups"];
+  repeated RawMetric metric_info = 13 [json_name="-"];
+  repeated Graph graphs = 14;
+}
+
+message TestMetadata {
```

Comment:
I do not see where this is used, please remove it until then

Reply:
This is referenced by the Response proto as a key/value pair for test_metadata